### PR TITLE
Code cleanup and docs update

### PR DESCRIPTION
**Pull request by sinhaashish**
1. Removed unused exception  `InvalidObjectPrefixException`.
2. API.md updated with `removeAllBucketNotification`

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -4691,7 +4690,6 @@ public void putObject(String bucketName, String objectName, InputStream stream,
    * @throws InternalException           upon internal library error
    * @throws InvalidArgumentException    upon invalid value is passed to a method.
    * @throws InvalidResponseException    upon a non-xml response from server
-   * @see #putObject(String bucketName, String objectName, String fileName)
```
Comment (Praveenrajmani) [inline api/src/main/java/io/minio/MinioClient.java]: Why do we remove this?

File: docs/API.md [hunk 2]
```diff
@@ -1008,6 +1008,74 @@ __Example__
     }
 ```
 
+  <a name="removeAllBucketNotification"></a>
```
Comment (Praveenrajmani) [inline docs/API.md]: An extra space in the head

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (sinhaashish) [reply api/src/main/java/io/minio/MinioClient.java]: Here we were referring to the overloaded function which is already deprecated.

File: docs/API.md (hunk 2 above)
Comment (sinhaashish) [reply docs/API.md]: done.

**Pull request merged**